### ISSUE TITLE
Suggestion: Dynamic Lights for Resource Packs

### ISSUE THREAD

**Issue by Lovyxia** [sp614x/optifine#1748]
This is my suggestion: The ability to make **specific textures** in resource packs turn whatever item they represent into dynamic light sources.

The point of this is to allow for item variants such as custom swords (like a fire-sword) in resource packs for use with adventure maps and the like so that you can have increased immersion, without having to mod the game so much that it becomes incompatible with servers.

It could also be used to disable existing dynamic light sources if needed.

I know similar suggestions have been discussed already, but mine's a bit more specific and I'm suggesting this here because I'm not finding any other suitable mods for this (and I believe this feature could fit into optifine quite nicely, especially when combined with emissive textures).

This feature would really help a lot with the RPG-Style server I'm making with friends on "vanilla". (Spigot)

**Comment by MastaNub** [comment]
Also allow us to increase the range of illumination 🙏

**Comment by RacerN2O** [comment]
Yeah adjustable illumination strength would be nice alongside this, so that you can make items that barely glow etc.

**Comment by NeutronForge** [comment]
yeah that would be cool. I made a custom lava sword and want it to emit light, but i dont want the netherite sword to do that too. Would it be possible or is it already to do something like the custom named items thing? only with a specific name it would emit a specific light level?
